dom.go

```diff
@@ -349,7 +351,7 @@ func (h *HTML) reconcileChildren(prev *HTML) (pendingMounts []Mounter) {
 				pendingMounts = append(pendingMounts, m)
 			}
 			h.lastRenderedChild = nextChildRender
-			h.insertBefore(h.insertBeforeNode, nextChildRender)
+			h.appendChild(nextChildRender)
```

Comment:
As far as I could tell, there is no functional difference to doing `h.appendChild` versus `h.insertBefore` here because `h.insertBeforeNode` will always be `nil` here / it will just invoke `appendChild` internally. After all, this case does not handle keyed children / only handles new unkeyed children.

If I missed something here please lmk.

Reply:
If we're rendering inside a list with unkeyed children, we will have an insertion node here, this needs to remain an `insertBefore` call.

Reply:
Got it. I missed that. I will revert and document clearly why this is needed then :)